sw/host/opentitanlib/src/transport/ti50emulator/gpio.rs

```diff
@@ -76,80 +79,53 @@ impl TryFrom<u8> for GpioStateUpdate {
     }
 }
 
-/// Structure representing GPIO pin configuration state.
-#[derive(Clone, Copy, Serialize, Deserialize)]
-pub struct GpioConfiguration {
-    pin_mode: Option<PinMode>,
-    pull_mode: PullMode,
-    value: bool,
-}
-
-impl GpioConfiguration {
-    /// Update current GPIO configurations by applying changes
-    /// encoded in `buf`.
-    pub fn update(&mut self, buf: &[u8]) -> Result<()> {
-        if !buf.is_empty() {
-            for data in buf.iter() {
-                match GpioStateUpdate::try_from(*data)? {
-                    GpioStateUpdate::Value(value) => {
-                        self.value = value;
-                    }
-                    GpioStateUpdate::PinMode(mode) => {
-                        self.pin_mode = mode;
-                    }
-                    GpioStateUpdate::PullMode(mode) => {
-                        self.pull_mode = mode;
-                    }
-                }
-            }
-            return Ok(());
-        }
-        bail!("GPIO configuration buffer is empty");
-    }
-}
-
-impl fmt::Display for GpioConfiguration {
+impl fmt::Display for Logic {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        write!(f, "{:?} {} {}", self.pin_mode, self.pull_mode, self.value)
+        write!(f, "{}", std::str::from_utf8(&[*self as u8]).unwrap())
     }
 }
 
-/// Multi value logic with drive strength representations.
-/// (Subset of IEEE 1164).
-enum Logic {
-    StrongZero,
-    StrongOne,
-    WeakZero,
-    WeakOne,
-    HiImpedance,
+impl Serialize for Logic {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(std::str::from_utf8(&[*self as u8]).unwrap())
```

Comment:
Since `Logic` implements `Display` (and rust will implement `ToString` for all `T: Display`):
```
serializer.serialize_str(&self.to_string())
```

Reply:
Done